### DISCUSSION TITLE
OEM firmware shortcoming when case battery is low/dead

### DISCUSSION THREAD

**Discussion by bitflung** [pine64/OpenPineBuds#50] [Ideas]
shortcoming of default firmware:
1. charged the buds and the case
2. used the buds over many days, dropping int the case to recharge each time
3. OBSERVATION: as per the manual, the buds power off automatically once dropped into the case
4. eventually the case battery discharged completely and i didn't realize it
5. OBSERVATION: with a dead case battery the buds do not detect they are in the case and they do not power off

As a user i found this confusing prior to realizing the issue was the dead case battery. i would use the buds, take them out, drop them in the case... and a few minutes later realize that my phone or laptop was still connected to the buds. initially i assumed the issue was that the contact between buds and case had failed due to dirt or debris... 

potential improvement in OpenPineBuds:
1. have the buds detect connection with the case even when the case battery is dead. perhaps we can detect VBAT even when it is lower than the charging threshold? if detected: POWER OFF
2. detect the State Of Charge (SoC) of the case battery whenever the buds are placed in the case; if SoC drops below some threshold (e.g. 20%) set a flag in the buds. on each power up when the buds play the audio indicating "power on" etc, they can additionally play audio to indicate "case battery is low".

with these two improvements a user will be informed when the case battery runs low and will continue to experience the expected buds behavior even after the case battery is fully depleted.

**Comment by Ralim** [comment]
The is no communication from the case to the buds sadly.
If you have a look at the buds schematics they can't directly tell they are in the case other than the charging signal. This is similar to a bunch of other designs as it simplifies the logic a lot.

Only thing I can think of is to test if we can detect the uart pins connecting and getting a load on them?

Otherwise could definitely add av button combo to force a power down?
